src/tests/ports/owners.spec.ts

```diff
@@ -0,0 +1,91 @@
+import { ISubgraphComponent } from '@well-known-components/thegraph-component'
+import { FetchOptions } from '../../ports/merger/types'
+import { createOwnersComponent } from '../../ports/owner/component'
+import {
+  IOwnerDataComponent,
+  OwnerFragment,
+  OwnersFilters,
+  OwnersSortBy,
+} from '../../ports/owner/types'
+
+let collectionSubgraphMock: ISubgraphComponent
+let getOwnersQueryMock: jest.Mock
+let ownersComponentMock: IOwnerDataComponent
+let queryFilters: FetchOptions<OwnersFilters, OwnersSortBy>
+
+describe('when fetching owners', () => {
+  beforeEach(() => {
+    getOwnersQueryMock = jest.fn()
+
+    collectionSubgraphMock = {
+      query: getOwnersQueryMock,
+    }
+
+    ownersComponentMock = createOwnersComponent({
+      subgraph: collectionSubgraphMock,
+    })
+
+    queryFilters = {
+      contractAddress: 'contractAddress',
+      itemId: 'itemId',
+      first: 10,
+      skip: 10,
+      sortBy: OwnersSortBy.ISSUED_ID,
+    }
+  })
+
+  afterEach(() => {
+    getOwnersQueryMock.mockRestore()
+  })
+
+  describe('and the request fails', () => {
+    beforeEach(() => {
+      getOwnersQueryMock.mockRejectedValueOnce(
+        new Error('itemId and contractAddress are neccesary params.')
```

Comment:
Would be better to change these tests to check the real failure of not having itemId or contractAddress. By setting those params in the `queryFilters` as undefined. One test for each.

Reply:
If you just want to test that the error is propagated another example would be better. Something simpler like "Query Error"